src/components/Sidebar.js

```diff
@@ -149,27 +179,51 @@ export default ({
     })
   }, [setSponsors])
 
-  if (isJudgingOpen) {
-    links.push({ location: '/judging', text: 'JUDGING' })
+  if (isSubmissionsOpen) {
+    links[1].push({ location: '/submission', text: 'Project Submission' })
   }
 
-  if (isSubmissionsOpen) {
-    links.push({ location: '/submission', text: 'SUBMISSION' })
+  if (isJudgingOpen) {
+    links[1].push({ location: '/judging', text: 'Peer Judging' })
   }
 
   if (user && user.admin) {
-    links.push({ location: '/judging/admin', text: 'JUDGING ADMIN' })
+    links[1].push({ location: '/judging/admin', text: 'Judging Admin' })
   }
 
   if (process.env.NODE_ENV !== 'production') {
-    links.push({ location: '/charcuterie', text: 'CHARCUTERIE' })
+    links[3].push({ location: '/charcuterie', text: 'CHARCUTERIE' })
   }
 
   if (isApplicationOpen) {
     // List the application as the last item on the menu
-    links.push({ location: '/application', text: 'APPLICATION' })
+    links[3].push({ location: '/application', text: 'APPLICATION' })
   }
 
+  const [renderLinks, setRenderLinks] = useState([])
+
+  useEffect(() => {
+    let renderArray = []
+
+    links.forEach((item, index) =>
+      renderArray.push(
+        <>
+          <CategoryHeader>{linksDict[index]}</CategoryHeader>
+          {item.map((link, i) => {
+            return (
+              <Link key={i} href={link.location} onClick={hideSidebarCallback}>
+                <StyledA selected={location === link.location}>{link.text}</StyledA>
+              </Link>
+            )
+          })}
+        </>
+      )
+    )
+
+    setRenderLinks(renderArray)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [links])
```

Comment:
Previously toggling a flag in Admin would update the sidebar without having to refresh. Wondering:
1. Is this behaviour still the same? and if so,
2. Is the reason the code here was taken out from the JSX because we're changing the data structure of the links?

Reply:
I used a design pattern I was familiar w/ for rendering array data BUT how it was before is much better so I've since changed the code. By extension, previous behavior should stay the same